Changed Client API docs structure

exec-astraea opened the pull request.

exec-astraea committed:
Moved Client API page into separate Studio Pro versions directories

WimJongeneel commented on content/en/docs/apidocs-mxsdk/apidocs/studio-pro-10/client-api.md:
```diff
@@ -23,9 +23,7 @@ The Mendix 10 Client API is available in Studio Pro 10.20.0 and above. For earli
 {{% /alert %}}
 
-## Legacy Client APIs {#legacy}
+## Legacy Client API {#legacy}
```
Should add anything about a deprecation and removal time line? (same of Mx 11 docs)

exec-astraea replied:
Yes, since we decided not to show any warnings in the browser, I'll add removal dates here.
Mendix 11 docs don't mention old APIs at all though.

exec-astraea committed:
Mention deprecation and removal of the legacy client API

weirdwater commented:
I like the restructure, but I have some notes on this MR:
- ~~I am missing updates to the index files of [Frontend](http://localhost:1313/apidocs-mxsdk/apidocs/apis-for-front-end/) and Studio Pro [11](http://localhost:1313/apidocs-mxsdk/apidocs/apis-for-studio-pro-11/)/[10](http://localhost:1313/apidocs-mxsdk/apidocs/apis-for-studio-pro-10/)/[9](http://localhost:1313/apidocs-mxsdk/apidocs/apis-for-studio-pro-9/)~~
- ~~Existing links to the old pages should continue to work. I am missing an [alias](https://gohugo.io/content-management/urls/#aliases) of the previous URL for the page that moved. (Unless there is tooling that does this automatically, @ConnorLand?)~~
- The design properties pages should be fairly straightforward to move and could already be included in this PR. 
- The widgets framework pages seem more involved, I will create ~~a PR targeting your branch~~ a [separate PR](https://github.com/mendix/docs/pull/9662).

EDIT: The index page is generated automatically, so no changes needed. My bad
EDIT2: No URLs changed, so you can disregard my remark about aliases and or redirects

ConnorLand committed:
uppercase the Mendix Client API

We lowercase client as a generic noun, but capitalize it when referring to Mendix's Client API. It's kind of weird, I know. @exec-astraea

ConnorLand committed:
Update client-api.md

ConnorLand committed:
Introduction text for bullet
